src/ILLink.RoslynAnalyzer/DynamicallyAccessedMembersAnalyzer.cs

```diff
@@ -53,33 +56,65 @@ public override void Initialize (AnalysisContext context)
 					}
 				}
 			});
-			// TODO: fix reporting for generic type substitutions. This should happen not only for method invocations,
-			// but for any reference to an instantiated method or type.
-			context.RegisterOperationAction (context => {
-				var invocationOperation = (IInvocationOperation) context.Operation;
-				ProcessInvocationOperation (context, invocationOperation);
-			}, OperationKind.Invocation);
+			context.RegisterSyntaxNodeAction (context => {
+				ProcessGenericParameters (context);
+			}, SyntaxKind.GenericName);
 		}
 
-		static void ProcessInvocationOperation (OperationAnalysisContext context, IInvocationOperation invocationOperation)
+		static void ProcessGenericParameters (SyntaxNodeAnalysisContext context)
 		{
-			if (context.ContainingSymbol.HasAttribute (RequiresUnreferencedCodeAnalyzer.RequiresUnreferencedCodeAttribute))
+			if (context.ContainingSymbol is not null && context.ContainingSymbol.HasAttribute (RequiresUnreferencedCodeAnalyzer.RequiresUnreferencedCodeAttribute)) {
```

Comment:
Good catch! Using `context.Node` instead of the `ContainingSymbol` to check for `IsInRequiresUnreferencedCodeAttributeScope` seems to work. Working on testing this fully and adding a test case.

Reply:
Chatted offline with @sbomer and continued using `ContainingSymbol` but excluding `INamedTypeSymbol`